src/support/dump-collection.ts

```diff
@@ -3,86 +3,234 @@ import path from 'path'
 
 import cli from 'cli-ux'
 import ndjson from 'ndjson'
+import { pickValues } from '../common'
+
+abstract class AbstractDumper {
+  protected collectionDir: string
+  protected filename?: string
+  protected writeStream?: fs.WriteStream
+  protected options: {scroll: string, size: number}
+
+  protected abstract get fileExtension(): string
+
+  constructor(
+    protected readonly sdk: any,
+    protected readonly index: string,
+    protected readonly collection: string,
+    protected readonly batchSize: number,
+    protected readonly destPath: string,
+    protected readonly query: any = {},
+  ) {
+    this.collectionDir = path.join(this.destPath, this.collection)
+    this.options = {
+      scroll: '1m',
```

Comment:
1 minute is very long for a scroll page TTL. It should be the time to write all the documents of the page (2000 by default, 10 000 maximum with Kuzzle/ES) in a file.

Reply:
This is code I just re-used from your implementation. I'll change it to 2 seconds.